recipes/ncvue/ncvue

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+"""
+usage: ncvue [-h] [-m missing_value] [netcdf_file]
+
+A minimal GUI for a quick view of netcdf files.
+
+positional arguments:
+  netcdf_file           netcdf file
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -m missing_value, --miss missing_value
+                        Set value to missing value (default: numpy.nan)
+
+
+Example command line:
+    ncvue MuSICA_out_2009.nc
+
+Written  Matthias Cuntz, Nov-Dec 2020
+"""
+from __future__ import absolute_import, division, print_function
+import numpy as np
+from ncvue import ncvue
+
+
+if __name__ == "__main__":
+
+    import argparse
+
+    miss = np.nan
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="""
+A minimal GUI for a quick view of netcdf files.""")
+    hstr  = 'Set value to missing value (default: numpy.nan)'
+    parser.add_argument('-m', '--miss', action='store', type=float,
+                        default=miss, dest='miss',
+                        metavar='missing_value', help=hstr)
+    parser.add_argument('ncfile', nargs='*', default=None,
+                        metavar='netcdf_file',
+                        help='netcdf file')
+
+    args   = parser.parse_args()
+    miss   = args.miss
+    ncfile = args.ncfile
+
+    del parser, args
+
+    if len(ncfile) > 0:
+        ncfile = ncfile[0]
+    else:
+        ncfile = ''
+
+    # This must be before any other call to matplotlib
+    # because it uses the TkAgg backend.
+    # This means, do not use --pylab with ipython.
+    ncvue(ncfile=ncfile, miss=miss)
```

Comment:
After moving the file into your python module as __main__.py, move this code into a main() function and then register this main() function as a console_script entry point.
Then pip will automatically wrap this entry point with a bash/bat script that will make the function available on the command line without calling the python interpreter directly.

https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

Reply:
@carterbox Created a gui_scripts entry_point. Thanks for the hint.